### PR TITLE
[FLINK-6103] LocalFileSystem rename() uses File.renameTo()

### PR DESCRIPTION
**Pull request by fpompermaier**
Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [x] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

- [x] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [x] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

### CODE REVIEW

File: flink-core/src/main/java/org/apache/flink/core/fs/local/LocalFileSystem.java [hunk 1]
```diff
@@ -262,8 +265,13 @@ public FSDataOutputStream create(
...
 		final File srcFile = pathToFile(src);
 		final File dstFile = pathToFile(dst);
-
-		return srcFile.renameTo(dstFile);
+		//Files.move fails if the destination directory doesn't exist
+		if(dstFile.getParentFile()!= null && !dstFile.getParentFile().exists()){
+			Files.createDirectories(dstFile.getParentFile().toPath());
+		}
+		//this throws an IOException if any error occurs
+		Files.move(srcFile.toPath(), dstFile.toPath(), StandardCopyOption.REPLACE_EXISTING);
```
Comment (StephanEwen) [inline flink-core/src/main/java/org/apache/flink/core/fs/local/LocalFileSystem.java]: If we want to preserve the old semantics, we should catch the exception and return `false` in that case.

Comment (fpompermaier) [inline flink-core/src/main/java/org/apache/flink/core/fs/local/LocalFileSystem.java]: So why is the method trowing an IOException?

Comment (StephanEwen) [inline flink-core/src/main/java/org/apache/flink/core/fs/local/LocalFileSystem.java]: I think that is inherited from the interface that is also used by the HdfsFileSystem. The outcomes are `true`(rename success), `false` (rename failed), and `IOExeption` (communication with filesystem failed).

**Comment by fpompermaier** [comment]
Does the current exception handling address your comments @StephanEwen?

**Comment by StephanEwen** [comment]
I think we don't need the logging in all of the cases. We typically avoid logging in "utility functions", which are missing the context if the exception is in fact a problem worth mentioning, of if it only adds noise to the log.

Other than that, I think its good.

**Comment by fpompermaier** [comment]
Anyone willing to merge this PR...?

**Comment by zentol** [comment]
@fpompermaier have you addressed the comment @StephanEwen regarding logging?

**Comment by fpompermaier** [comment]
Of course..but how should I handle them? should I catch just one exception? What should I log?

**Comment by StephanEwen** [comment]
I would do the following:
  - log nothing
  - Catch the errors that are regular "move failed" exceptions and return false.
     - `FileNotFoundException`
     - `DirectoryNotEmptyException`
     - `SecurityException`
  - Let all other `IOExceptions` bubble out

**Comment by fpompermaier** [comment]
Now it should be ok

**Comment by StephanEwen** [comment]
Thanks, I'll take it from here...

**Pull request closed without merging**